### ISSUE TITLE
Long "Time Remaining" after level reset

### ISSUE THREAD

**Issue by alltom** [davidsansome/tsurukame#455]
I reset my account to level 1 a few weeks ago. Now I'm on level 5 and I've noticed that my estimated "Time Remaining" for every level is 200–400 days, until I'm close to finishing, at which point is jumps down to 3–4 days.

In https://github.com/davidsansome/tsurukame/issues/418, @UInt2048 said "This should even out more after a few more levels" when someone had a similar issue, but I still don't see much difference 5 levels in. Maybe because my account is so old? I joined mid-2014.

**Comment by UInt2048** [comment]
@alltom Your issue _is_ a bug, Tsurukame should account for the reset but obviously isn't.

**Comment by UInt2048** [comment]
Right, that's correct. I assume it was more like a thousand days initially? It would take much longer to even out such a long gap.

Tsurukame really should account for resets though, so this issue should stay open.

**Comment by abstrctn** [comment]
To resolve this, I think there's a **product question to answer** about how much of a user's ["level progression"](https://docs.api.wanikani.com/20170710/#level-progressions) history to include when calculating the estimated time remaining for a level. (The issue seems specific to calculating [estimated times](https://github.com/davidsansome/tsurukame/blob/da1e5e762028d2fb549d34239a6390bc680cd8d8/ios/LevelTimeRemainingItem.swift#L74-L79), when locked kanji at the current level remain.)

Currently, **the estimate seems based on a user's [entire level progression history](https://github.com/davidsansome/tsurukame/blob/da1e5e762028d2fb549d34239a6390bc680cd8d8/ios/LevelTimeRemainingItem.swift#L90-L94)**, excluding any level progression with a `timeSpentCurrent` [value of 0](https://github.com/davidsansome/tsurukame/blob/da1e5e762028d2fb549d34239a6390bc680cd8d8/ios/WaniKaniAPI/Sources/WaniKaniAPI/ProtobufExtensions.swift#L400-L409). This exclusion seems to exist so that any abandoned level is excluded from the estimation data - which seems correct given that many level resets would occur after a long period of inactivity, skewing the estimate. But this method of excluding individual preogressions still seems to give users higher than expected estimates. I haven't confirmed this with a local build, but it seems likely that **the outlier datapoints increasing the estimate are from the period before a user's last level reset**.

I'm curious to get other opinions, but my instinct would be that any estimate after a level reset should be based solely on level progressions since that reset. In other words, the estimate could take the list of ["all level progressions"](https://github.com/davidsansome/tsurukame/blob/da1e5e762028d2fb549d34239a6390bc680cd8d8/ios/LevelTimeRemainingItem.swift#L90-L94), and then truncate any previous levels beginning with and including the last known reset level (the last level progression with a populated `abandoned_at` value).
